### PR TITLE
feat: PC-13482 Add static validation from SDK

### PR DESCRIPTION
**Pull request by BSski**
## Motivation

Validating resources at the stage of `terraform plan` instead of `terraform apply` will shorten the developer's feedback loop.

## Summary

Added validation to the `terraform plan` stage for all resources besides `BudgetAdjustment` and `SLO`.

## Release Notes

Added validation to the `terraform plan` stage for all resources besides `BudgetAdjustment` and `SLO`.

### CODE REVIEW

File: nobl9/resource_agent.go [hunk 1]
```diff
@@ -115,6 +116,19 @@ func agentSchema() map[string]*schema.Schema {
 	return agentSchema
 }
 
+//nolint:unparam
```
Comment (nieomylnieja) [inline nobl9/resource_agent.go:119]: instead of adding this nolint you can just use `_` for parameter name

File: nobl9/resource_service.go [hunk 2]
```diff
@@ -104,6 +105,19 @@ func unmarshalService(d *schema.ResourceData, objects []v1alphaService.Service)
...
+	if diags.HasError() {
+		return diagsToSingleError(diags)
+	}
+	errs := manifest.Validate([]manifest.Object{service})
+	if errs != nil {
+		return formatErrorsAsSingleError(errs)
+	}
+	return nil
+}
+
```
Comment (nieomylnieja) [inline nobl9/resource_service.go:120]: placeholder comment: please remove `nolint: unused` comments from functions which are now used

File: nobl9/resource_agent.go (hunk 1 above)
Comment (BSski) [reply nobl9/resource_agent.go]: Done. Also removed a few other unnecessary ones. Thanks!

File: nobl9/resource_service.go (hunk 2 above)
Comment (BSski) [reply nobl9/resource_service.go:119]: Done. Removed a few other unnecessary ones too. Thanks!

**Review by nieomylnieja — APPROVED** [review]
nicely done, great that it works!

File: nobl9/resource_budgetadjustment_test.go [hunk 3]
```diff
@@ -10,11 +10,11 @@ import (
 )
 
 func TestAcc_Nobl9BudgetAdjustments(t *testing.T) {
-	futureTime := time.Now().AddDate(0, 0, 1)
+	futureTime := time.Now().AddDate(0, 0, 1).Format("2006-01-02T15:04:05Z")
```
Comment (nieomylnieja) [inline nobl9/resource_budgetadjustment_test.go:13]: why do you need to use some custom time format? :) just use `time.RFC3339` and if you need to make sure the string is in UTC just call `.UTC()` on `time.Time` object

**Review by nieomylnieja — CHANGES_REQUESTED** [review]

File: nobl9/resource_budgetadjustment_test.go (hunk 3 above)
Comment (BSski) [reply nobl9/resource_budgetadjustment_test.go:13]: Right, thanks!

**Review by nieomylnieja — APPROVED** [review]
lgtm!

**Pull request merged**